Sources/TensorFlow/Operators/LinearAlgebra.swift

```diff
@@ -143,6 +143,24 @@ public func trace<T: TensorFlowNumeric>(_ matrix: Tensor<T>) -> Tensor<T> {
     return matrix.diagonalPart().sum(squeezingAxes: -1)
 }
 
+/// Computes the determinant of one or more square matrices.
+/// 
+/// - Parameter matrix: A tensor of shape `[..., M, M]`.
+/// - Returns: A tensor containing the determinants of all input submatrices.
+@inlinable 
+func det<T:TensorFlowFloatingPoint>(_ matrix: Tensor<T>) -> Tensor<T> {
```

Comment:
Although the check makes sense, is there any way to avoid these precondition overheads o.O

Reply:
I think `precondition` itself is lightweight because the `det` operation and any other linalg operation are much more expensive in terms of clock time.

Reply:
`-Onone` performance regressions due to adding `precondition`s are not a big concern.
`precondition`s can be disabled with `-Ounchecked`, e.g. `swift build -c release -Xswiftc -Ounchecked`.

[This blog post](https://blog.krzyzanowskim.com/2015/03/09/swift-asserts-the-missing-manual/) has more info on which optimization modes disable assertions/preconditions.

All of these details are in the PR description of https://github.com/tensorflow/swift-apis/issues/517.

Reply:
Not adding a `precondition` for now sounds fine.

There are other operations that expect `Tensor` arguments to have specific shapes that are also missing `precondition`s; `precondition`s can be added later at once.